src/core/plugin/luapi_application.h

```diff
@@ -327,6 +384,263 @@ static int applib_layerAction(lua_State* L) {
     return 1;
 }
 
+static int addStrokeHelper(lua_State* L, Stroke* stroke, Layer* layer) {
+    //    Plugin* plugin = Plugin::getPluginFromLua(L);
+    //    Control* ctrl = plugin->getControl();
+    //    PageRef const& page = ctrl->getCurrentPage();
+    //    Layer* layer = page->getSelectedLayer();
+    //    Stroke* stroke = new Stroke();
+
+    // discard any extra arguments passed in
+    lua_settop(L, 1);
+    luaL_checktype(L, 1, LUA_TTABLE);
+
+    // Make sure we have enough points to form a stroke
+    if (stroke->getPointCount() >= 2) {
+        // Get the rest of the attributes. Default to something if none is provided
+        // TODO: (willnilges) Test this. Is this robust? Should we be doing this one field at a time?
+        lua_getfield(L, 1, "width");
+        lua_getfield(L, 1, "color");
+        lua_getfield(L, 1, "fill");
+        lua_getfield(L, 1, "tool");
+        lua_getfield(L, 1, "lineStyle");
+        double width = luaL_optnumber(L, -5, 1.5);
+        long long int color = luaL_optinteger(L, -4, 0);
+        int fill = luaL_optinteger(L, -3, -1);
+        const char* tool = luaL_optstring(L, -2, "pen");
+        const char* style = luaL_optstring(L, -1, "");
+        lua_pop(L, 5);
+
+        stroke->setWidth(width);
+        stroke->setColor(Color(color));
+        stroke->setFill(fill);
+        if (strcmp("eraser", tool) == 0) {
+            stroke->setToolType(STROKE_TOOL_ERASER);
+        } else if (strcmp("pen", tool) == 0) {
+            stroke->setToolType(STROKE_TOOL_PEN);
+        } else if (strcmp("highlighter", tool) == 0) {
+            stroke->setToolType(STROKE_TOOL_HIGHLIGHTER);
+        } else {
+            g_warning("%s", FC(_F("Unknown stroke type: \"{1}\", assuming pen") % tool));
+        }
+        LineStyle lineStyle = StrokeStyle::parseStyle(style);
+        stroke->setLineStyle(lineStyle);
+        if (strcmp("", style) != 0 && strcmp("dash", style) != 0 && strcmp("dashdot", style) != 0 &&
+            strcmp("dot", style) != 0)
+            g_warning("%s", FC(_F("Unknown style: \"{1}\", assuming solid") % style));
+        layer->addElement(stroke);
+        stroke = nullptr;
+        return 1;
```

Comment:
If you make the return type `void` you will also have to change the return statements (here and below).
```suggestion
        return;
```